### ISSUE TITLE
[bug] `no-leaked-event-listener` false positive when using React Native `BackHandler`

### ISSUE THREAD

**Issue by SuperKXT** [Rel1cx/eslint-react#1323]
### Describe the bug

`BackHandler.addEventListener` returns a `remove` method, but the rule complains when it is used.

```tsx
import { BackHandler } from "react-native";

useEffect(() => {
  const { remove }  = BackHandler.addEventListener("hardwareBackPress", onBackPress);
  return () => {
    remove();
  }
});
```

### Platform and versions

```Text
* @eslint-react/eslint-plugin v2.3.4
* Node.js v24.11.1
```

### Stack trace

```Text

```
